kernel/model/model.go

```diff
@@ -393,146 +386,199 @@ type Action interface {
 	Execute(m *Model) error
 }
 
+func NewRunContext(label *Label, model *Model) RunContext {
+	return &runContext{
+		label: label,
+		model: model,
+		runId: fmt.Sprintf("%d", info.NowInMilliseconds()),
+	}
+}
+
+type RunContext interface {
+	GetModel() *Model
+	GetLabel() *Label
+	GetId() string
+}
```

Comment:
What if we just called this `model.Run`?

Reply:
:+1: